src/app/user/user.ts

```diff
@@ -40,11 +46,18 @@ export class User extends Entity {
     }).toString();
 
     this.password = {'hash': hash, 'salt': cryptSalt, 'iterations': cryptIterations, 'keysize': cryptKeySize};
+
+    // update encrypted nextcloud password
+    this.blobPasswordEnc = CryptoJS.AES.encrypt(this.blobPasswordDec, password).toString();
   }
 
   public checkPassword(givenPassword: string): boolean {
     // compares given password to the stored one of this user
     // therefore hashes the given password string and compares it with the sored hash
+    if  (this.hashPassword(givenPassword) === this.password.hash) {
```

Comment:
I think writing a getter for `blobPasswordDec` would be better than doing the decryption here. The method name `checkPassword` gives the impression that there will be no side effects

Reply:
removed that part; made decryptBlobPassword() public and used it in session service.